### PR TITLE
Feat/separator component

### PR DESCRIPTION
**Pull request by MaciejMichalski92**

### CODE REVIEW

File: src/components/Separator/index.tsx [hunk 1]
```diff
@@ -0,0 +1 @@
+export { Separator as default } from './Separator';
```
Comment (kubaf13) [inline src/components/Separator/index.tsx:1]: add 
`export type { SeparatorProps } from './Separator.types'`

**Review by kubaf13 — CHANGES_REQUESTED** [review]

File: src/components/Separator/Separator.types.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export type SeparatorType = 'SMALL' | 'MEDIUM';
+
+export interface SeparatorProps {
+  type: SeparatorType;
+}
```
[the remark this answers is not in the document]
Comment (MaciejMichalski92) [reply src/components/Separator/Separator.types.ts:5]: I think there is no need to add an additional type in React function for TS. If we are using a .tsx file we know that we will use tsx component. TS right now is very good at type inference, and I don't think we need to add this ts rule to this config. Additional work and time for nothing. And for custom functions that are not React component we will add the return type and check it on CR. What do you think?

File: src/components/Separator/Separator.styled.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import styled from 'styled-components';
+import type { SeparatorType } from './Separator.types';
+
+export const StyledSeparator = styled('div')<{ type: SeparatorType }>(
```
[the remark this answers is not in the document]
Comment (MaciejMichalski92) [reply src/components/Separator/Separator.styled.ts:4]: why?

File: src/components/Separator/Separator.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+// explicite path for storybook component
+import separatorImg from '../../../public/images/separator-line.svg';
+import { StyledSeparator } from './Separator.styled';
+import type { SeparatorProps } from './Separator.types';
+
+export const Separator = ({ type }: SeparatorProps) => (
```
[the remark this answers is not in the document]
Comment (MaciejMichalski92) [reply src/components/Separator/Separator.tsx:6]: why change the type declaration for the type alias? We won't be reusing this type SeparatorType anywhere else, so we can do interface for props and let TS do the type inference for returning the React Function.

File: src/components/Separator/Separator.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+// explicite path for storybook component
+import separatorImg from '../../../public/images/separator-line.svg';
+import { StyledSeparator } from './Separator.styled';
+import type { SeparatorProps } from './Separator.types';
```
[the remark this answers is not in the document]
Comment (MaciejMichalski92) [reply src/components/Separator/Separator.tsx:4]: as above

File: src/components/Separator/index.tsx (hunk 1 above)
Comment (MaciejMichalski92) [reply src/components/Separator/index.tsx:1]: do we need to export it?

File: src/components/Separator/Separator.tsx [hunk 6]
```diff
@@ -0,0 +1,10 @@
+// explicite path for storybook component
+import separatorImg from '../../../public/images/separator-line.svg';
```
[the remark this answers is not in the document]
Comment (MaciejMichalski92) [reply src/components/Separator/Separator.tsx:2]: I think if we will delete the Storybook, I can go back to the '@' path (which didn't worked for storybook server)

**Review by kubaf13 — APPROVED** [review]
git

File: src/fables_library/pl/library.ts
Comment (lastDunadan) [inline src/fables_library/pl/library.ts:1]: Pracuję nad nowym wariantem tego pliku. Jakbyś miał konflikty, odrzuć swoje zmiany i zaakceptuj moje. Jak nie masz, to możesz to przepchnąć i mój komit wszystko naprawi/nadapisze.

**Review by lastDunadan — APPROVED** [review]

**Pull request merged**